docs/source/working_with_aiida/caching.rst

```diff
@@ -4,45 +4,92 @@
+ 1. Place the following ``cache_config.yml`` file in your ``.aiida`` configuration folder
+    in order to enable caching for your AiiDA profile (here called ``aiida2``):
 
-    In [1]: from __future__ import print_function
+    .. code:: yaml
 
-    In [2]: from aiida.orm import Str
+        aiida2:
+          default: True
 
-    In [3]: n1 = Str('test string')
+ 2. Run ``verdi -p aiida2 rehash`` in order to compute hashes of all nodes in your existing database.
 
-    In [4]: n1.store()
-    Out[4]: u'test string'
+From this point onwards, when you launch a new calculation, AiiDA will compute its hash (depending both on the type of calculation and its inputs, see :ref:`caching_matches`).
+If another calculation with the same hash is found to be already present in the database, AiiDA will reuse its results without repeating the actual calculation.
 
-    In [5]: n2 = Str('test string')
+.. _fig_caching:
+.. figure:: include/images/caching.png
+  :align: center
+  :height: 350px
 
-    In [6]: n2.store(use_cache=True)
-    Out[6]: u'test string'
+  When reusing a cached calculation, AiiDA links up to the input nodes as usual, and copies both the calculation node **C** and its outputs **D3**.
+  While this sketch shows **C'** reuse the input nodes of **C'**, it is only the *content* of the input nodes (their hash) that matters.
 
-    In [7]: print('UUID of n1:', n1.uuid)
-    UUID of n1: 956109e1-4382-4240-a711-2a4f3b522122
+In order to ensure that the provenance graph with and without caching is the same,
+AiiDA creates both a new calculation node and a copy of the output data nodes as shown in :numref:`fig_caching`.
 
-    In [8]: print('n2 is cached from:', n2.get_extra('_aiida_cached_from'))
-    n2 is cached from: 956109e1-4382-4240-a711-2a4f3b522122
+.. note:: Caching is **not** implemented at the WorkChain/workfunction level (see :ref:`caching_limitations` for details).
+
+
+.. _caching_matches:
+
+How are nodes hashed?
+---------------------
+
+By default (see also :ref:`devel_controlling_caching`), the hash of a Data node is computed from:
+
+* all attributes of the node, except the ``_updatable_attributes`` and ``_hash_ignored_attributes``
+* the ``__version__`` of the package which defined the node class
+* the content of the repository folder of the node
+* the UUID of the computer, if the node is associated with one
+
+The hash of a :class:`~aiida.orm.ProcessNode` will, on top of this, include the hashes of any of its input ``Data`` nodes.
+
+Once a node is stored in the database, its hash is stored in the ``_aiida_hash`` extra, and this extra is used to find matching nodes.
+If a node of the same class with the same hash already exists in the database, this is considered a cache match.
+
+Use the :meth:`~aiida.orm.nodes.Node.get_hash` method to check the hash of any node.
+
+In order to figure out why a calculation is *not* being reused, the :meth:`._get_objects_to_hash() <aiida.orm.nodes.Node._get_objects_to_hash>` may be useful:
```

Comment:
For consistency maybe change to `:meth:`~aiida.orm.nodes.Node._get_objects_to_hash``